File: core/play/src/main/scala/play/api/routing/HandlerDef.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.routing

/**
 * Information about a `Handler`, especially useful for loading the handler
 * with reflection.
 */
case class HandlerDef(
    classLoader: ClassLoader,
    routerPackage: String,
    controller: String,
    method: String,
    parameterTypes: Seq[Class[?]],
    verb: String,
    path: String,
    comments: String = "",
    modifiers: Seq[String] = Seq.empty
) extends play.routing.HandlerDef

File: core/play/src/main/java/play/routing/HandlerDef.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.routing;

import java.util.List;
import play.libs.Scala;
import scala.collection.Seq;

public abstract class HandlerDef {
  public abstract ClassLoader classLoader();

  public abstract String routerPackage();

  public abstract String controller();

  public abstract String method();

  protected abstract Seq<Class<?>> parameterTypes();

  public abstract String verb();

  public abstract String path();

  public abstract String comments();

  protected abstract Seq<String> modifiers();

  public List<Class<?>> getParameterTypes() {
    return Scala.asJava(parameterTypes());
  }

  public List<String> getModifiers() {
    return Scala.asJava(modifiers());
  }
}
